examples/run/cuda/bfield_cuda.cu

```diff
@@ -0,0 +1,625 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include <cmath>
+#include <fstream>
+#include <iostream>
+#include <limits>
+
+#include "cuda/utils/definitions.hpp"
+
+/*
+ * This structure contains all the information we need to use the magnetic field
+ * on the GPU, at least in this simple context. It contains the texture in GPU
+ * memory, the offsets in each direction (because texture indexing starts at
+ * zero, but we can have negative coordinates), and the total size of the
+ * magnetic field volume.
+ */
+struct MagneticField {
+    /*
+     * This is a CUDA object referring to a texture, the details are opaque and
+     * we are not allowed to know what is inside.
+     */
+    cudaTextureObject_t texture;
+
+    /*
+     * The offsets for coordinates in the B-field volume.
+     */
+    float offset_x, offset_y, offset_z;
```

Comment:
I think we have to use `scalar` instead of `float`

Reply:
Possibly texture doesnt work with double?

Reply:
Correct, this only works with single-precision indices.